src/events/messageCreate/train.ts

```diff
@@ -0,0 +1,24 @@
+import { ArgsOf, Discord, On } from "discordx";
+import { DEV_BOT_ID } from "../../constants.js";
+
+const activeTrains = new Set<String>();
+@Discord()
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+abstract class Train {
+	@On({ event: "messageCreate" })
+	private async handleTrain([msg]: ArgsOf<"messageCreate">) {
+		const msgs = await msg.channel.messages.fetch({ limit: 2 });
+		if (msgs.every((e) => e.content === "")) return;
+		if (
+			!activeTrains.has(msg.channelId) &&
+			msgs.at(0).content === msgs.at(1).content &&
+			msgs.at(0).author.id !== msgs.at(1).author.id
+		) {
+			activeTrains.add(msg.channelId);
+			if (msgs.some((e) => e.author.id === DEV_BOT_ID)) return;
```

Comment:
think we can use `client.botId` here instead of DEV_BOT_ID or the BDBOT id to be more general

Reply:
Other than this LGTM

Reply:
I would agree but in my testing client.botId was just returning the string 'bot' for some reason

Reply:
Oh that's bizarre. I'll see if there's a way to grab the real id then

Reply:
I see the same thing lol very strange. Probably a bug with a library we're using. Can instead use `client.user.id`